mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAOptimizer.scala

```diff
@@ -451,10 +451,10 @@ final class OnlineLDAOptimizer extends LDAOptimizer {
       }
       Iterator((stat, gammaPart))
     }
-    val statsSum: BDM[Double] = stats.map(_._1).reduce(_ += _)
```

Comment:
This doesn't seem right because the first arg is modified in-place, which violates reduce contract. It should be an aggregate (or treeAggregate) instead.